tests/integrationv2/test_happy_path.py

```diff
@@ -61,6 +61,9 @@ def test_s2n_server_happy_path(managed_process, cipher, provider, curve, protoco
         assert bytes("Actual protocol version: {}".format(expected_version).encode('utf-8')) in results.stdout
         assert random_bytes in results.stdout
 
+        if provider is OpenSSL:
```

Comment:
Why do we only do this check for OpenSSL? Wouldn't we want to check this for every provider?

Reply:
We can only control the cipher through non-s2n providers, so this should actually read `if provider is not S2N`, updated.